### DISCUSSION TITLE
Unable to use Component Blocks in Keystone 6. Getting ERR_PACKAGE_PATH_NOT_EXPORTED error

### DISCUSSION THREAD

**Discussion by vishwa795** [keystonejs/keystone#9082] [Questions] [answered]
Use the keystone boilerplate repository. In the post schema, add a component block exactly as mentioned in the [docs](https://keystonejs.com/docs/guides/document-fields#component-blocks). 

Application should run fine.

<img width="903" alt="Screenshot 2024-03-31 at 12 18 15 AM" src="https://github.com/keystonejs/keystone/assets/54198625/e5cb997e-7dc9-4a63-ac66-d829fa0e0fbf">
<img width="552" alt="Screenshot 2024-03-31 at 12 19 18 AM" src="https://github.com/keystonejs/keystone/assets/54198625/b2294e8a-6bac-411e-8f0e-65b83f2da54f">

```
✨ Starting Keystone
(node:21488) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/declarations/src/component-blocks' is not defined by "exports" in /Users/vishwa/Desktop/work/synapse/node_modules/@keystone-6/fields-document/package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:416:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:669:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at /Users/vishwa/Desktop/work/synapse/node_modules/next/dist/server/require-hook.js:54:36
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at Module.mod.require (/Users/vishwa/Desktop/work/synapse/node_modules/next/dist/server/require-hook.js:64:28) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
error Command failed with exit code 1.
```

node version: 17.0.0
Latest keystone version

**Comment by borisno2** [comment]
Hi @vishwa795,
The error suggests you are trying to import something from `@keystone-6/fields-document/dist/declarations/src/component-blocks` but that is not defined in the exports of the package, what are you importing from that location?

**Comment by vishwa795** [comment]
Ah, my bad the import was hapenning from wrong location.

I guess vscode automatically imported from the wrong location.
However, Im getting a diferrent error now.
![image](https://github.com/keystonejs/keystone/assets/54198625/0fcffb7b-edfc-4329-93e6-aec77d5947ff)

This is how i'm using the component blocks ^^

Getting the following error:

```
Module not found: Can't resolve '/Users/vishwa/Desktop/work/curio/apps/component-blocks.tsx'
   8 | import * as view5 from "@keystone-6/core/fields/types/timestamp/views";
   9 | import * as view6 from "@keystone-6/fields-document/views";
> 10 | import * as view7 from "/Users/vishwa/Desktop/work/curio/apps/component-blocks.tsx";
  11 | 
  12 | var adminConfig = {};
  13 | 
```

**Comment by borisno2** [comment]
Your `views: path.resolve("../component-blocks.tsx")` on line 32 is being converted to `/Users/vishwa/Desktop/work/curio/apps/component-blocks.tsx` which can't be found, so your path of `../component-blocks.tsx` is not correct.

**Comment by vishwa795** [comment]
<img width="1326" alt="Screenshot 2024-04-03 at 5 41 46 PM" src="https://github.com/keystonejs/keystone/assets/54198625/b4d5f17c-0eea-4a0b-a160-5ea6c2475f7a">


```
ui: {
          views: path.resolve(__dirname, "..", "src/component-blocks.tsx"),
      }
```

I got it working with the above snippet. But when i try to use 

```
ui: {
          views: "../component-blocks.tsx",
        }
```

It fails with the Module not found error. In the screen shot attached above you can see that I'm referencing my parent directory for component blocks. Ideally `../component-blocks.tsx` should have also worked.

EDIT: Maybe it's trying to look into .keystone folder:

<img width="1161" alt="image" src="https://github.com/keystonejs/keystone/assets/54198625/548182c2-59cd-4ff8-8db7-263ac79ae6b0">

```
ui: {
          views: path.resolve(__dirname, "component-blocks.tsx"),
        }
```
 Location referenced for above output.

**Reply by dcousens** [reply]
@vishwa795 maybe https://github.com/keystonejs/keystone/blob/main/examples/custom-field-view/schema.ts#L27 helps as an example reference?

**Reply by gautamsi** [reply, marked as answer]
@vishwa795 keep in mind that `views: "../component-blocks.tsx"` the path is supposed to be relative to the package.json in the repo and not relative to the file where you are using this line.
